### DISCUSSION TITLE
Cannot invoke "java.nio.file.Path.getFileName()" because "p" is null

### DISCUSSION THREAD

**Discussion by heshanthenura** [jreleaser/jreleaser#1606] [Q&A] [answered]
Anyone know how to fix this?
I'm using maven
![image](https://github.com/jreleaser/jreleaser/assets/75155192/cf1fb08a-9f40-451b-86ef-199fe4777131)

**Comment by aalmiray** [comment]
You'll have to give us more details to find out what's going on. From the looks of it a file/artifact is missing. Please past the contents of `target/jreleaser/trace.log` after editing out any private information found in that file. If there were to be no stacktrace in that file then run Maven once more with the `-X` flag activated to get the stacktrace and paste it here.

**Reply by heshanthenura** [reply]
Here is my trace.log
[trace.log](https://github.com/jreleaser/jreleaser/files/14423466/trace.log)

**Reply by aalmiray** [reply]
The NPE occurs in this piece of code https://github.com/jreleaser/jreleaser/blob/119983a46b4cedadf6924a353a8c18b5beec8428/core/jreleaser-model-impl/src/main/java/org/jreleaser/model/spi/deploy/maven/Deployable.java#L74-L98 in line 84 which would be the last in the following snippet

```
            Path p = Paths.get(path);
            this.version = p.getFileName().toString();
            p = p.getParent();
            this.artifactId = p.getFileName().toString();
```

This strikes me as odd as the input `path` should point to the full location where the file to be deployed may be found. What are the contents of the `target/staging-deploy` directory? If you happen to have the `tree` command then please run `tree target/staging-deploy`.

**Reply by heshanthenura** [reply]
This is the content
![image](https://github.com/jreleaser/jreleaser/assets/75155192/fbbee6f7-4adc-4615-93b1-e0046dc9c1ef)
![image](https://github.com/jreleaser/jreleaser/assets/75155192/c67072f2-3217-47c1-91bd-4472a33e45e6)

**Reply by aalmiray** [reply, marked as answer]
I believe `EasyJava-0.1.zip` is the problem. That file should not be there. Its directory structure does not follow the expectations. Either remove that file _or_ place it in the correct directory which is the same as the other files.

I'd recommend posting the zip file as a release asset instead of deploying to Maven Central.

**Reply by heshanthenura** [reply]
fixed the issue. Thanks for the help
